### ISSUE TITLE
Enabling a cookie type from preferences should hide notification bar

### ISSUE THREAD

**Issue by Grsmto** [chiiya/haven#125]
Hi!

If a user enables a cookie type from the preferences panel (if user clicks directly on "settings" instead of accepting or declining all), the notification bar still shows up all the time. Even though the user explicitly chose his preferences.
I think the bar should hide since we can suppose the user has chosen at that point.
I had a look and I think whenever `enableCookies` is called, it should set every untouched cookies to their default value (`false` in that case).

What do you think?

**Comment by chiiya** [comment]
Yea, that's correct. Is that an issue for your use case? I suppose we could add an opt-out flag for that functionality if it leads to some unintentional behaviour.

**Comment by Grsmto** [comment]
I guess that makes sense in term of UX that the changes you do are not applied until you hit "save". If there is a save button, implicitly it means that there are 2 states.
However that would make the code more complex as we would need to add an internal state to save the user changes and apply them on change...

In the meanwhile, I did the change that I suggested on my fork, so that the bar hides whenever the user switch a toggle.
